hrjobcontract/js/test/services/contract-pay.service.spec.js

```diff
@@ -15,8 +15,8 @@ define([
       $httpBackend = _$httpBackend_;
       $rootScope = _$rootScope_;
 
-      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond(MockContract.contractPayment);
-      $httpBackend.whenGET(/action=get&entity=HRJobPay/).respond({});
+      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond(200);
+      $httpBackend.whenGET(/action=get&entity=HRJobPay/).respond(MockContract.contractPayment);
```

Comment:
why was this changed and how come it was not breaking before?

Reply:
the only explanation i can give is that 1.6 and 1.7 are stricter with tests, while 1.5 let a lot of things slide, even if wrong.

You can definitely tell that the "before" was wrong (returning a mocked contract payment on `entity=HRJobContract` instead of `entity=HRJobPay`), but still didn't cause a blip on 1.5. As soon as the version was bumped, you correctly got the error.

I didn't go much deeper than that tbh, at that point I'd have to go reverse engineer what angular does, and didn't seem worth it